src/common/tse.c

```diff
@@ -642,12 +644,19 @@ tse_task_post_process(tse_task_t *task)
 		if (task_tmp->dt_result == 0 && !dtp_tmp->dtp_no_propagate)
 			task_tmp->dt_result = task->dt_result;
 
+		dsp_tmp = dtp_tmp->dtp_sched;
+		diff_sched = dsp != dsp_tmp;
+
+		if (diff_sched) {
+			D_MUTEX_UNLOCK(&dsp->dsp_lock);
+			D_MUTEX_LOCK(&dsp_tmp->dsp_lock);
```

Comment:
Hope the dtp_tmp will not be added to other dependency during waiting for the dsp_tmp->dsp_lock.

Reply:
looks like that even dtp_tmp will be added another dep task here, it won't cause problem because "dtp_tmp->dtp_dep_cnt--" below will not reach zero then.